### PR TITLE
qgscamerapose: Fix pitch min value to avoid rotation computation error

### PR DESCRIPTION
**Pull request by ptitjano**
This prevents a bug in QgsCameraPose::updateCamera when updating
camera rotation.
Indeed, with a mPitchAngle < 0.2, QQuaternion::fromEulerAngles(
mPitchAngle, mHeadingAngle, 0 ) will return bad rotation angle.

This is fixed is Qt6, but it has not been backported to Qt5.
See: https://bugreports.qt.io/browse/QTBUG-72103

cc @benoitdm-oslandia

### CODE REVIEW

**Comment by ptitjano** [comment]
@wonder-sk Could you take a look at it please? This should fix an issue mentioned in the [3D axis PR](https://github.com/qgis/QGIS/pull/48109#issuecomment-1145007215).

File: src/3d/qgscamerapose.cpp [hunk 1]
```diff
@@ -62,8 +62,11 @@ void QgsCameraPose::setPitchAngle( float pitch )
   // prevent going over the head
   if ( pitch > 180.0f )
     mPitchAngle = 180.0f;
-  else if ( pitch < 0.0f )
-    mPitchAngle = 0.0f;
+  else if ( pitch < 0.2f )
```
Comment (lbartoletti) [inline src/3d/qgscamerapose.cpp:65]:
```suggestion
  else if ( pitch < 0.2f )
```

Maybe use `std::clamp` from algorithm include:

`std::clamp(mPitchAngle, 0.2f, 180.0f)`

Comment (ptitjano) [reply src/3d/qgscamerapose.cpp]: Good idea. Applied. Thanks

**Review by lbartoletti — APPROVED** [review]

**Comment by uclaros** [comment]
@ptitjano This has fixed the 3D axis issue only partially. The axes are now ok when viewing the scene from top, but the issue still appears when the scene is viewed from the bottom.

**Pull request merged**